cmd/mistryd/worker_pool.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/skroutz/mistry/pkg/types"
+)
+
+// WorkerPool implements a fixed-size pool of worker goroutines that can be sent
+// build jobs and communicate their result
+type WorkerPool struct {
+	// the fixed amount of goroutines that will be handling running jobs
+	MaxWorkers int
+	// the maximum backlog of pending job requests. if exceeded, sending new work
+	// to the pool will return an error
+	QueueSize int
+	queue     chan workItem
+}
+
+// WorkResult contains the result of a job, either a buildinfo or an error
+type WorkResult struct {
+	BuildInfo *types.BuildInfo
+	Err       error
+}
+
+// FutureWorkResult is a WorkResult that has not yet become available
+type FutureWorkResult struct {
+	resultQueue <-chan WorkResult
+}
+
+// Wait waits for the result to become available and returns it
+func (future FutureWorkResult) Wait() (WorkResult, error) {
+	result, ok := <-future.resultQueue
+	if !ok {
+		return WorkResult{}, fmt.Errorf("Failed to read from result channel")
+	}
+	return result, nil
+}
```

Comment:
We have no guarantee that it will not fail so we should handle it. The only way it can fail right now is if a bug is introduced. I considered `panic()`ing but I thought it's too much.

Even if panic() is acceptable at this API level, it is not acceptable to crash the mistry service from it, so I would have to write a panic error handler that returns an internal server error